### PR TITLE
Feature/use google cloud storage

### PR DESCRIPTION
**Pull request by jaensen**

### CODE REVIEW

File: src/environment.ts [hunk 1]
```diff
@@ -116,14 +116,16 @@ export class Environment {
...
-    if (
-      !process.env.BUCKET_ENDPOINT ||
-      !process.env.BUCKET_KEY ||
-      !process.env.BUCKET_SECRET
-    ) {
+    console.log(`* Checking GOOGLE_CLOUD_STORAGE_CREDENTIALS ...`);
+    if (!this.googleCloudStorageCredentials?.project_id) {
       errors.push(
-        `The BUCKET_ENDPOINT, BUCKET_KEY or BUCKET_SECRET environment variable is not set.`
+        `The GOOGLE_CLOUD_STORAGE_CREDENTIALS environment variable contains a invalid json object.`
```
Comment (almereyda) [inline src/environment.ts]:
```suggestion
        `The GOOGLE_CLOUD_STORAGE_CREDENTIALS environment variable contains an invalid JSON object.`
```

or maybe even (see below)

```suggestion
        `The GCS_CREDENTIALS environment variable contains an invalid JSON object.`
```

File: src/httpHandlers/post/upload.ts [hunk 2]
```diff
@@ -26,59 +24,36 @@ export const uploadPostHandler = async (req: Request, res: Response) => {
...
     const mimeType = req.body.mimeType;
-    const bytes = req.body.bytes;
+    const bytes = Buffer.from(req.body.bytes.replace(/^data:image\/\w+;base64,/, ""), "base64");
 
-    const saveResult = await saveImageToS3(fileName, bytes, mimeType);
+    let storage = new Storage({
+      credentials: Environment.googleCloudStorageCredentials
+    });
+
+    const uploadResult = await storage.bucket("local-api-server-ztq5-asia-southeast2-bucket").file(objectKey).save(bytes, {
```
Comment (almereyda) [inline src/httpHandlers/post/upload.ts]: My suggestion would be not to hard code an eventual bucket here, but to make it configurable via an environmental variable called `GCS_BUCKET` or alike.

File: src/httpHandlers/post/upload.ts [hunk 3]
```diff
@@ -26,59 +24,36 @@ export const uploadPostHandler = async (req: Request, res: Response) => {
...
+      metadata: {
+        contentType: mimeType,
+      },
+    });
+
+    // Use the Google cloud storage sdk to get the file's public url
+    const publicUrl = await storage.bucket("local-api-server-ztq5-asia-southeast2-bucket").file(objectKey).getSignedUrl({
```
Comment (almereyda) [inline src/httpHandlers/post/upload.ts]: See above.

**Comment by almereyda** [comment]
Thanks for the timely implementation.

From an operative perspective it could be nice to have the new variable a little easier to read. `GOOGLE_CLOUD_STORAGE_CREDENTIALS` is very broad for the eye and not easily to skim in one glance. (Many characters are not concerned with the domain object they describe, but just expose a brands of a vendor.)

My suggestion is to name it `GCS_CREDENTIALS` instead.

File: src/httpHandlers/post/upload.ts (hunk 2 above)
Comment (jaensen) [reply src/httpHandlers/post/upload.ts]: ouch. good find :)

**Pull request merged**